helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -97,10 +97,28 @@ public ParticipantManager(HelixManager manager, HelixZkClient zkclient, int sess
   }
 
   /**
-   * Handle new session for a participang.
+   * Handles a new session for a participant.
+   *
+   * @deprecated
+   * This is deprecated. Instead, please use {@link #handleNewSession(String)} which takes care of
+   * potential session race condition issue.
+   *
    * @throws Exception
    */
+  @Deprecated
   public void handleNewSession() throws Exception {
+    handleNewSession(null);
+  }
+
+  /**
+   * Handles a new session for a participant. The new session's id has to be passed in, as it is
+   * required in {@link #createLiveInstance(String)} to prevent ephemeral node creation from session
+   * race condition: ephemeral node is created by an expired or unexpected session.
+   *
+   * @param sessionId the new session's ID
+   * @throws Exception if any exception occurs
+   */
+  public void handleNewSession(final String sessionId) throws Exception {
     joinCluster();
```

Comment:
Before joinCluster, we can check if the sessionId is the same as current ZkClient sessionId. If it is not, no need to do anything except printing a log. 
This is an outdated one by the time of execution.

Reply:
can _zkClient.getSessionId() == 0? 

I think they can. We should do nothing here then.